docs/operate/get-started/other-hardware/dependencies.md

```diff
@@ -0,0 +1,251 @@
+      }
+      s.camera = camera
+
+      return s, nil
+    }
+   ```
+
+1. You can now call API methods on the dependency resource within your module, for example:
+
+   ```go {class="line-numbers linkable-line-numbers"}
+   img, imgMetadata, err := s.camera.Image(ctx, utils.MimeTypeJPEG, nil)
+   ```
+
+{{% alert title="Note on reconfiguration" color="note" %}}
+
+Most Go modules use `resource.AlwaysRebuild` within the `<module-name><resource-name>` struct, which means that the resource rebuilds every time the module is reconfigured.
+
+The steps above use `resource.AlwaysRebuild`.
+If you need to maintain the state of your resource, see [(Optional) Create and edit a `Reconfigure` function](/operate/get-started/other-hardware/#implement-the-component-api).
+
+{{% /alert %}}
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### Optional dependencies
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+If your module has optional dependencies, the steps are the same as for required dependencies, except that your `validate_config` method can treat the dependency as optional by returning an empty list if the dependency is not configured:
+
+```python {class="line-numbers linkable-line-numbers"}
+@classmethod
+def validate_config(cls, config: ComponentConfig) -> Sequence[str]:
+    deps = []
+    fields = config.attributes.fields
+    if "camera_name" not in fields:
+        self.logger.info(
+          "camera_name not configured, using no camera")
+    else:
+        if not fields["camera_name"].HasField("string_value"):
+            raise Exception("camera_name must be a string")
+        deps.append(fields["camera_name"].string_value)
+    return deps
+```
+
+Be sure to handle the case where the dependency is not configured in your API implementation as well.
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+
+If your module has optional dependencies, the steps are the same as for required dependencies, except that your `Validate` method can treat the dependency as optional by returning an empty list if the dependency is not configured:
+
+```go {class="line-numbers linkable-line-numbers"}
+func (cfg *Config) Validate(path string) ([]string, error) {
+  var deps []string
+  if cfg.CameraName == "" {
+    logger.Info("camera_name not configured, using no camera")
+    return nil, nil
+  }
+  if reflect.TypeOf(cfg.CameraName).Kind() != reflect.String {
+    return nil, errors.New("camera_name must be a string")
+  }
+  deps = append(deps, cfg.CameraName)
+  return deps, nil
+}
+```
+
+Be sure to handle the case where the dependency is not configured in your API implementation as well.
+
+{{% /tab %}}
+{{< /tabs >}}
+
+{{% hiddencontent %}}
+There is not currently an SDK method to directly access configuration attributes of dependencies in Python or Go, but in Python it is possible to use `get_robot_part` to return information including the whole configuration of a machine part, and then access the configuration attributes of the dependency from there.
```

Comment:
what does this mean? I think we would want to discourage users from looking at configs for other resources

Reply:
We got a chatbot question about this specifically (from Sean Yu) and the bot answer was incorrect so was hoping to fix it. This hidden content short code means it’s not visible on the page, only to the chatbot if someone specifically asks. But if we don’t want people to do this at all should I just change this to say “you can’t, period”?

Reply:
ok, I guess in that case this is fine - we should definitely discourage it tho